clippy_lints/src/methods/filter_next.rs

```diff
@@ -26,15 +42,30 @@ pub(super) fn check<'tcx>(
         if filter_snippet.lines().count() <= 1 {
             let iter_snippet = snippet(cx, recv.span, "..");
             // add note if not multi-line
-            span_lint_and_sugg(
-                cx,
-                FILTER_NEXT,
-                expr.span,
-                msg,
-                "try this",
-                format!("{iter_snippet}.find({filter_snippet})"),
-                Applicability::MachineApplicable,
-            );
+            span_lint_and_then(cx, FILTER_NEXT, expr.span, msg, |diag| {
+                let (applicability, pat) = if let Some(id) = path_to_local(recv)
+                    && let Some(hir::Node::Pat(pat)) = cx.tcx.hir().find(id)
+                    && let hir::PatKind::Binding(BindingAnnotation(_, Mutability::Not), _, ident, _) = pat.kind
+                {
+                    (Applicability::Unspecified, Some((pat.span, ident)))
```

Comment:
It can no longer be a machine applicable suggestion if it needs to be made mutable because of cases like these:
```rs
let (Ok(f) | Err(f)) = ...;
f.filter(...).next();
```
There are multiple bindings that would need to be made mutable, but `cx.tcx.hir().find(id)` only returns one pattern.

`iter_skip_next` is doing this as well, so I think having it be unspecified is fine.
https://github.com/rust-lang/rust-clippy/blob/ce0a48a9b66148fdd96fddf585f9e11850861a47/clippy_lints/src/methods/iter_skip_next.rs#L23-L35

Reply:
I gues we can walk the parent to find all the bindings, but that's also not simple. This is good enough for now